lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -599,8 +683,13 @@ SerializedModuleLoaderBase::findModule(ImportPath::Element moduleID,
     }
   };
 
-  auto result = forEachModuleSearchPath(
-      Ctx,
+  SmallVector<std::string, 3> InterestingFilenames = {
+      (moduleName + ".framework").str(),
+      genericBaseName.getName(file_types::TY_SwiftModuleInterfaceFile),
+      genericBaseName.getName(file_types::TY_SwiftModuleFile)};
```

Comment:
Should we also check for `TY_PrivateSwiftModuleInterfaceFile` here?

Reply:
I checked that it wasn’t necessary because both places that can consume the private swift module interface bail out if the public one doesn’t exist.

https://github.com/apple/swift/blob/2ef512722ae8bbcc7dff272a9db99e1dd8a33752/lib/Frontend/ModuleInterfaceLoader.cpp#L1108
bails out if the public file doesn’t exist here
https://github.com/apple/swift/blob/2ef512722ae8bbcc7dff272a9db99e1dd8a33752/lib/Frontend/ModuleInterfaceLoader.cpp#L1097

---

https://github.com/apple/swift/blob/2ef512722ae8bbcc7dff272a9db99e1dd8a33752/lib/Serialization/ModuleDependencyScanner.cpp#L61
asserts that the public file exists here
https://github.com/apple/swift/blob/2ef512722ae8bbcc7dff272a9db99e1dd8a33752/lib/Serialization/ModuleDependencyScanner.cpp#L56

Reply:
That's interesting I didn't realize we had this restriction in place. Thank you for looking into it.

Reply:
Just to be safe (also for the future if we decide to lift that restriction), I added the private swiftinterface file as well.